src/pages/onboarding/index.tsx

```diff
@@ -0,0 +1,227 @@
+import { useEffect, useState } from "react";
+import { motion, AnimatePresence } from "framer-motion";
+import Head from "next/head";
+import { useRouter } from "next/router";
+
+const questions = [
+    {
+        id: "username",
+        question: "Choose Your Username",
+        type: "text",
+        required: true,
+    },
+    {
+        id: "role",
+        question: "What do you do?",
+        type: "radio",
+        options: ["Student", "Working Professional"],
+        required: true,
+    },
+    {
+        id: "usage",
+        question: "How would you use the Platform?",
+        type: "checkbox",
+        options: ["Learning Tech", "Building Projects", "Interview Prep", "Job Search"],
+        required: true,
+    },
+    {
+        id: "contact",
+        question: "Your Contact No (with Country Code)",
+        type: "text",
+        required: true,
+    },
+];
+
+export default function Onboarding() {
+    const router = useRouter();
+    const [step, setStep] = useState(0);
+    const [formData, setFormData] = useState<any>({});
+    const [redirectTo, setRedirectTo] = useState("/dashboard");
+
+    const current = questions[step];
+    const isLast = step === questions.length - 1;
+
+    useEffect(() => {
+        if (router.query?.redirectTo) {
+            setRedirectTo(router.query.redirectTo as string);
+        }
+    }, [router.query]);
+
+    const handleChange = (id: string, value: any) => {
+        setFormData({ ...formData, [id]: value });
+    };
+
+    const next = () => {
+        if (!isLast) {
+            setStep(step + 1);
+        } else {
+            console.log("Onboarding Data:", formData);
+            localStorage.setItem("isOnboarded", "true");
+            router.push(redirectTo);
+        }
+    };
+
+    const prev = () => {
+        if (step > 0) setStep(step - 1);
+    };
+
+    return (
+        <>
+            <Head>
+                <title>Onboarding | TBE</title>
+            </Head>
```

Comment:
Not following our Coding practices. We never use this component in our Pages. Check once